### PR TITLE
Stac 3402- Allow https on agent6

### PR DESCRIPTION
**Pull request by vishalxebia**
### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

**Comment by Voronenko** [comment]
I think, proposed ruins a bit original test.
More adequate seems extracting that in separate scenario
check
https://molecule.readthedocs.io/en/latest/

File: test/molecule-role/molecule/default/tests/test_agent.py [hunk 1]
```diff
@@ -42,13 +42,24 @@ def wait_for_check_successes():
...
+                line):
+            continue
+
+        if re.search(
+                "x509: certificate signed by unknown authority",
+                line):
+            continue
+
+        if re.search(
+                "Too many errors for endpoint \'https://testagent.com/*",
```
Comment (Voronenko) [inline test/molecule-role/molecule/default/tests/test_agent.py]: Too many errors is ok?

File: test/molecule-role/molecule/default/prepare.yml [hunk 2]
```diff
@@ -133,12 +148,29 @@
...
+      become: true
+      lineinfile:
+        path: /etc/hosts
+        line: "{{ hostvars['receiver']['ansible_host'] }} testagent.com"
     - name: Install stackstate-agent
       shell: "curl -o- {{agent_repo_url}}/install.sh | bash"
       environment:
         STS_API_KEY: "API_KEY"
-        STS_URL: "http://{{ hostvars['receiver']['ansible_host'] }}:7077/stsAgent"
+        STS_URL: "https://testagent.com/stsAgent"
```
Comment (Voronenko) [inline test/molecule-role/molecule/default/prepare.yml]: I find that change errorneous.  Suggesting to split into separate scenario

File: test/molecule-role/molecule/default/prepare.yml [hunk 3]
```diff
@@ -75,14 +75,23 @@
...
+      with_items:
+        - docker.nginx
+        - nginx-signed.crt
+        - nginx-signed.key
+        - docker-compose.yml
     - name: Run Docker compose
       command: docker-compose up -d
       args:
         chdir: /home/ubuntu/
+    - name: Restart Nginx Service
```
Comment (TizianoPerrucci) [inline test/molecule-role/molecule/default/prepare.yml]: why do we need to restart nginx ?

File: test/molecule-role/molecule/default/prepare.yml [hunk 4]
```diff
@@ -108,6 +117,12 @@
         name: nc
         state: present
       when: ansible_distribution == "Fedora"
+    - name: Install libselinux-python in Fedora
```
Comment (TizianoPerrucci) [inline test/molecule-role/molecule/default/prepare.yml]: why do we need libselinux on the agent ?

File: test/molecule-role/molecule/default/prepare.yml [hunk 5]
```diff
@@ -133,12 +148,35 @@
...
+      lineinfile:
+        path: /etc/hosts
+        line: "{{ hostvars['receiver']['ansible_host'] }} test-stackstate-agent.sts"
     - name: Install stackstate-agent
       shell: "curl -o- {{agent_repo_url}}/install.sh | bash"
       environment:
         STS_API_KEY: "API_KEY"
         STS_URL: "http://{{ hostvars['receiver']['ansible_host'] }}:7077/stsAgent"
         CODE_NAME: "{{agent_current_branch}}"
+    - name: Replace STS URL to https URL
```
Comment (TizianoPerrucci) [inline test/molecule-role/molecule/default/prepare.yml]: The `STS_URL` is passed as environment variable to the installation script in the above task, can't we change it there directly ?

File: test/molecule-role/molecule/default/tests/test_agent.py [hunk 6]
```diff
@@ -42,10 +42,18 @@ def wait_for_check_successes():
...
         # TODO: Update event endpoint should get rid of this error,
         # once STAC-2500 is fixed
+        if re.search(
+                "Error code \"400 Bad Request\" received while sending " +
+                "transaction to \"https://.*/stsAgent/intake/",
+                line):
+            continue
+
+        # Before restarting the logs contain error for previous sts_url
         if re.search(
```
Comment (TizianoPerrucci) [inline test/molecule-role/molecule/default/tests/test_agent.py]: We will never see http again in the logs, why keeping this assertion ?

File: test/molecule-role/molecule/default/prepare.yml [hunk 7]
```diff
@@ -133,12 +148,35 @@
...
+      become: true
+      replace:
+        path: /etc/stackstate-agent/stackstate.yaml
+        regexp: 'sts_url: http://.*:7077/stsAgent'
+        replace: 'sts_url: https://test-stackstate-agent.sts/stsAgent'
+    - name: Skip SSL Validation
```
Comment (TizianoPerrucci) [inline test/molecule-role/molecule/default/prepare.yml]: Should we add the parameter `SKIP_SSL_VALIDATION` to the installation script ?
So we don't need this ansible task and the next one

File: test/molecule-role/molecule/default/prepare.yml (hunk 5 above)
Comment (vishalxebia) [reply test/molecule-role/molecule/default/prepare.yml]: yeah changing the URL here leads to `x509: certificate signed by unknown authority` error which will be in the logs until restarted with `SKIP_SSL_VALIDATION: true` flag.

File: test/molecule-role/molecule/default/prepare.yml (hunk 4 above)
Comment (vishalxebia) [reply test/molecule-role/molecule/default/prepare.yml]: I was getting this error "Aborting, target uses SELinux but python bindings (libselinux-python) aren’t installed" and found this blog `https://medium.com/@audun.nes/aborting-target-uses-selinux-but-python-bindings-libselinux-python-arent-installed-f6b9d4542130` which says it comes during molecule test.

File: test/molecule-role/molecule/default/tests/test_agent.py (hunk 6 above)
Comment (vishalxebia) [reply test/molecule-role/molecule/default/tests/test_agent.py]: before changing the URL and SSL flag in the conf file and restarting agent, the log contains `http` log.

File: test/molecule-role/molecule/default/prepare.yml (hunk 7 above)
Comment (vishalxebia) [reply test/molecule-role/molecule/default/prepare.yml]: yeah true. added in the install script.

**Pull request merged**